### PR TITLE
Add notes about TEP numbers

### PR DESCRIPTION
**Pull request by afrittoli**
Until we automate this https://github.com/tektoncd/community/issues/178
provide guidelines to handle TEP numbers and avoid confusion / churn.

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>

/cc @vdemeester @ImJasonH @bobcatfish @abayer

### CODE REVIEW

File: teps/README.md [hunk 1]
```diff
@@ -70,6 +70,12 @@ The TEP `OWNERS` are the **main** owners of the following projects:
 
 ## Merging TEPs
 
+- When creating a new TEP, pick a number for it, equal to the maximum
+  TEP available in the repo and open PRs. The TEP number shall be set
+  in the TEP file name, TEP title and PR title:
+  - file name in the format `teps/<XXXX>-<tep-title>.md`
+  - title in the format `# TEP-XXXX: <tep-title>`
+  - PR title in the format `TEP-XXXX: <tep-title>`
```
Comment (bobcatfish) [inline teps/README.md:78]: does this fix the problem? i think folks pretty much need to pick the number right before merging

OR we get faster at merging things with "proposed" status vs leaving them open (not sure how we continue discussing them in that case? in the merged PR maybe?)

Comment (vdemeester) [reply teps/README.md:78]: So I think how it works in k8s is, at some point, a number is assigned (by one of the approvers) and we go from there. I am fine adding the number as a follow-up too.

Comment (afrittoli) [reply teps/README.md:78]: We can also implement it this way, similar to k8s style. I don't mind either way, as long as we fix the issue for authors of picking a number.

File: hack/tep-number.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+set -o nounset
+set -o pipefail
+
+# Obtain the list of TEPs from the repo + the list of open PRs
+# whose title starts with TEP-NNNN
+LAST_TEP=$(
+  {
+      curl https://api.github.com/repos/tektoncd/community/pulls 2>/dev/null | \
+      jq -r '.[] | select(.state == "open") | .title ' | grep -oP 'TEP-[0-9]{4}' | cut -d'-' -f2 & \
+      ls teps/ | grep -oP '[0-9]{4}';
```
Comment (chhsia0) [inline hack/tep-number.sh:30]: `-P` is not supported by BSD grep (e.g., on macOS):
```suggestion
      jq -r '.[] | select(.state == "open") | .title ' | grep -oE 'TEP-[0-9]{4,4}' | cut -d'-' -f2 & \
      ls teps/ | grep -oE '[0-9]{4,4}';
```

File: hack/tep-number.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  } | sort | tail -1)
+
+printf "%04g" "$((10#$LAST_TEP + 1))"
```
Comment (chhsia0) [inline hack/tep-number.sh:33]:
```suggestion
printf "%04d\n" "$((10#$LAST_TEP + 1))"
```

**Comment by bobcatfish** [comment]
I think I was making a bigger deal out of trying to find the "perfect" way to do this than is necessary, I think what you proposed is effective and let's do it!

File: hack/tep-number.sh (hunk 3 above)
Comment (afrittoli) [reply hack/tep-number.sh:33]: I did not add a new line on purpose, I prefer the output to be just the string

File: hack/tep-number.sh (hunk 2 above)
Comment (afrittoli) [reply hack/tep-number.sh:30]:
```
brew install grep
alias grep=ggrep
```

Comment (chhsia0) [reply hack/tep-number.sh:30]: Yeah that would work. Just wondering if there's any concern to use a syntax that works with BRE and don't require GNU grep ;)

File: hack/tep-number.sh (hunk 3 above)
Comment (chhsia0) [reply hack/tep-number.sh:33]: It's just that without the newline the shell prompt will be in the same line as the output. But it's not a big deal.

**Comment by bobcatfish** [comment]
In API working group today we talked about merging this as-is so that ppl have something they can use to help them pick TEP numbers and we'll iterate from there

/lgtm

**Pull request merged**